Docs process tweaks: update PR template & CODEOWNERS (#4591)
withastro/astro: .github/PULL_REQUEST_TEMPLATE.md

## Docs

<!-- Could this affect a user’s behavior? We probably need to update docs! -->

> <!-- If adding docs or unsure, uncomment the next line: -->

Reviewer: ```suggestion
<!-- If adding docs or unsure, uncomment the next line: -->
```

To me as a non-native speaker, the first part of the sentence ("If adding docs or unsure") is unclear. Can we rephrase that to make it crystal clear what the condition for uncommenting the following line is?

Author: Does this feel clearer?

```suggestion
<!-- If docs will be needed or you’re not sure, uncomment the next line: -->
```

Reviewer: Yes, that's much better! Thank you :)

> <!-- Is this a visible change? You probably need to update docs! -->

Reviewer 2: ```suggestion
<!-- Could this change affect a user's behavior? Ping @withastro/maintainers-docs and let them know! -->
```
@sarah11918 Thoughts on this tweak?

Reviewer 3: You read my mind, @Jutanium! 🙌 

The key idea for me is: 
- will a user (have to) do something differently now? (Docs can decide if that conflicts with current documentation - no need to for devs to worry about that!)

This includes might a user WANT TO do something differently now? e.g. We have a better simpler way because of a new feature or bug fix that we would now recommend instead, even if this isn't a breaking change and the old way still works.

I'm happy with the shortest "trigger" for people that gets them thinking about this when deciding whether or not to ping docs. Especially when thinking about external PRs received, it needs to be "enough" for someone outside the normal process to make that connection. I think this does that!

Reviewer 4: I was going to say, you could go a step further here and make the instructions more clear for the author:

```
<!-- DON'T DELETE THIS SECTION! If no docs added, explain why.-->
<!-- If adding docs (or unsure!) uncomment the next line: -->
<!-- /cc @withastro/maintainers-docs for feedback! -->
```

Or something to that effect!

Author: Pushed a commit combining these suggestions — let me know what you all think!
